### DISCUSSION TITLE
Is there anyway to delete offline Runner Scale Set from GitHub Repo/Org?

### DISCUSSION THREAD

**Discussion by ryaugusta** [actions/actions-runner-controller#3035] [FAQs] [answered]
I've been testing the `gha-runner-scale-set` and noticed that when tearing down & reinstalling the scale-set that I still have the old runners stuck "offline" in the GitHub UI. I ran into some issues redeploying the runners if the scale set name was the same for some reason. I did a complete tear down and  changed the name of the runner-scale-set and deployed which got the runners up but I now have two "Runner scale sets" listed in the GitHub UI for my repo - only one is active. 

What if I want to delete this "offline" set? I can't find a way to remove this in the UI or documentation.

<img width="864" alt="Screenshot 2023-10-30 at 11 03 28 AM" src="https://github.com/actions/actions-runner-controller/assets/26639389/e1116f50-23e1-4e20-a94d-21913741034a">

**Comment by Link-** [comment, marked as answer]
All unused runner scale sets will be purged automatically after 24 hours. At the moment we do not expose APIs  (through `api.github.com`) to manage them. It's definitely in our backlog though.

**Reply by ryaugusta** [reply]
Thanks @Link- , This was my as I did things out of order when uninstalling but that is good to know that they will be automatically purged after 24 hours 🙌

What I experienced is a result of removing the `gha-runner-scale-set-controller` before removing the `gha-runner-scale-set`. Removing the `runner-scale-set` first allows the controller to correctly delete the necessary resources. Found this info [here](https://github.com/actions/actions-runner-controller/issues/2781#issuecomment-1658256533)

Since I did things out of order, I had to manually delete resources (`serviceaccount`, `secrets`, `rolebinding`, `role`, etc..) to redeploy runners with the name I was trying to remove - then `helm uninstall` the release so the controller could properly remove all of the resources. Doing this fixed the issue and cleaned everything up properly.

**Reply by Link-** [reply]
yeah I understand, we do the clean-up normally except for out of order scenarios. Hopefully soon you'll have the capacity to manage runner scale sets more directly.

**Comment by krupakar1329** [comment]
@Link-  "All unused runner scale sets will be purged automatically after 24 hours." Does this statement holds true?
i have a self hosted runner scaleset created at the repository level. Now i want to move it to organization level. but facing the issue
had issue communicating with Actions backend: No runner scale set found with identifier 1."}.
Once the self hosted runner scaleset is offiline, is it deleted by Github after 24 hours?

**Reply by Link-** [reply]
Yes

**Comment by maeghan-porter** [comment]
I have quite a few runner-scale-sets at the moment that have def been gone for more than 24 hours and they're still hanging around saying offline.

**Reply by Link-** [reply]
Opening a support ticket for investigation is the appropriate next step

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.

**Comment by kingphil** [comment]
I figured out a "work-around" for this, and went through the process of removing old/stale arc-runner instances that were listed as "Offline" (for far greater than 24 hours).

The answer is to redo the deployment, presumably Helm chart install, with the same name to the same runner group. It can be a minRunner and maxRunner of 0; this is just to get the runner instance to change from "Offline" to "Online".

Then, manually delete the autoscalingrunnersets (this removes the arc-runner instance), followed by uninstalling the Helm chart.
```
kubectl -n arc-runners delete autoscalingrunnersets.actions.github.com arc-alpha
helm -n arc-runners uninstall arc-alpha
```
